src/utils.js

```diff
@@ -477,6 +481,55 @@ async function encryptSecrets(publicKey, secrets) {
 	}
 }
 
+// Function to replace escaped variables with placeholders
+const replaceEscapedVariables = content => {
+	const placeholderMap = {};
+	const newContent = content.replace(/\\+[$](([a-zA-Z]+)|([{][a-zA-Z]+[}]))/g, matched => {
+		const slashCount = (matched.match(/\\/g) || []).length;
+		if (slashCount % 2 !== 0) {
+			const placeholder = `__PLACEHOLDER_${Math.random().toString(36).substr(2, 9)}__`;
+			const newValue = reduceConsecutiveBackslashes(matched);
+			placeholderMap[placeholder] = newValue;
+			return placeholder;
+		} else {
+			return reduceConsecutiveBackslashes(matched);
+		}
+	});
+	return [newContent, placeholderMap];
```

Comment:
My suggestion, dont make it an array. Return an object instead.

```suggestion
	return {
	    secrets: newContent,
	    placeholderMap
	};
```